### DISCUSSION TITLE
Ternary Logic Block

### DISCUSSION THREAD

**Discussion by johnmrichey** [pybricks/support#1563] [General]
Newbie question here, how do I build this Ternary Logic Block for this step of the program example please?
<img width="521" alt="Screenshot 2024-03-22 at 19 50 54" src="https://github.com/pybricks/support/assets/63120157/cdf00ee1-53ac-4b59-b8d3-909f559fef6a">

**Comment by laurensvalk** [comment]
It's on the logic tab:

![image](https://github.com/pybricks/support/assets/12326241/0e52b561-adb1-4ca5-bbb6-d794df44ad3d)

You can intuitively read it as a normal sencence: So the value becomes "the first value if the condition is true, else the other value".

You can put any values, like a number you type in, or a color such as green, red (from the input tab).

The condition can be anything as well. In the example you're quoting, it's a boolean value that is either True or False.

**Reply by johnmrichey** [reply]
I can't figure out how to change the input from a value as a number to color from the input tab.  Which does one select from the input tab for color please?
<img width="648" alt="Screenshot 2024-03-23 at 06 41 02" src="https://github.com/pybricks/support/assets/63120157/b3d505ca-e493-4161-a4c8-32bf74700228">

**Reply by johnmrichey** [reply]
I figured it out, thanks.  Just have to drag the color to the beginning of the block and then put it into the value field.
<img width="724" alt="Screenshot 2024-03-23 at 06 53 09" src="https://github.com/pybricks/support/assets/63120157/841b195d-7860-47e3-9fa6-2e4e46b80384">

**Reply by laurensvalk** [reply]
Glad you figured it out. I should have probably shown that step in the video :)

**Comment by johnmrichey** [comment]
To get a color as a value into a logic block, first grab the color sensor block from Input and place it on the workspace, then grab the color variable out of the block and drag it to the workspace by itself, then you can place it into the logic block as the variable.
<img width="385" alt="Screenshot 2024-03-23 at 08 13 06" src="https://github.com/pybricks/support/assets/63120157/5f41ac76-ed7e-4f16-9020-72425c594dfe">
<img width="485" alt="Screenshot 2024-03-23 at 08 12 17" src="https://github.com/pybricks/support/assets/63120157/326bd54b-6b25-4077-b3ad-f53a34a56b52">

**Comment by johnmrichey** [comment]
I added in the Controller to the program so it only runs when the Button A on the xbox controller is pressed.
<img width="421" alt="Screenshot 2024-03-23 at 08 33 40" src="https://github.com/pybricks/support/assets/63120157/f07a5f4e-6a92-4775-9f70-87c5e56c7552">
